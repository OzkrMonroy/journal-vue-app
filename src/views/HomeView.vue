<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
  </div>
  <button @click="goToDayBook" class="btn btn-primary">Primary</button>
  <button @click="goToDayBook" class="btn btn-secondary">Secondary</button>
  <button @click="goToDayBook" class="btn btn-success">Success</button>
</template>

<script>
export default {
  name: "HomeView",
  components: {},
  methods: {
    goToDayBook() {
      this.$router.push({ name: "daybook" });
    },
  },
};
</script>
