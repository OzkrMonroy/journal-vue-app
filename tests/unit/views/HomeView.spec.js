import { shallowMount } from "@vue/test-utils";
import HomeView from "@/views/HomeView";

describe("HomeView tests", () => {
  test("should render correctly", () => {
    const wrapper = shallowMount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should redirect when a button is clicked", () => {
    const mockRouter = {
      push: jest.fn(),
    };
    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
