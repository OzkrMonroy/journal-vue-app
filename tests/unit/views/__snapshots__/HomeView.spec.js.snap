// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView tests should render correctly 1`] = `
<div class="home"><img alt="Vue logo" src=""></div>
<button class="btn btn-primary">Primary</button>
<button class="btn btn-secondary">Secondary</button>
<button class="btn btn-success">Success</button>
`;
